### PR TITLE
Test to verify header size in Beast front end using s3cmd

### PR DESCRIPTION
**Pull request by anrao19**
Signed-off-by: Anuchaithra Rao <anrao@redhat.com>
Test to verify header size in Beast front end using s3cmd
Log : http://magna002.ceph.redhat.com/ceph-qe-logs/Anuchaithra/test_header_size.txt
updated log : http://magna002.ceph.redhat.com/ceph-qe-logs/Anuchaithra/header_size.txt

### CODE REVIEW

File: rgw/v2/tests/s3cmd/configs/test_header_size.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# script: test_header_size.py
+config:
+  user_count: 1
+  bucket_count: 1
```
Comment (udaysk23) [inline rgw/v2/tests/s3cmd/configs/test_header_size.yaml:4]: I think script will ignore second user or second bucket if we increase `user_count` or `bucket_count` from 1 to more

Comment (anrao19) [reply rgw/v2/tests/s3cmd/configs/test_header_size.yaml:4]: Made changes

**Review by udaysk23 — DISMISSED** [review]
lgtm

File: rgw/v2/tests/s3cmd/test_header_size.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+from v2.lib import resource_op
+from v2.lib.exceptions import RGWBaseException, TestExecError
+from v2.lib.s3.write_io_info import BasicIOInfoStructure, IOInfoInitialize
+from v2.lib.s3cmd import auth as s3_auth
+from v2.tests.s3cmd import reusable as s3cmd_reusable
+from v2.utils import utils
+from v2.utils.log import configure_logging
+from v2.utils.test_desc import AddTestInfo
+
+log = logging.getLogger()
```
Comment (psathyan) [inline rgw/v2/tests/s3cmd/test_header_size.py:39]: could you add the filename... this helps during triage... `log = logging.getLogger(__name__)`

File: rgw/v2/tests/s3cmd/test_header_size.py [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    basic_io_structure = BasicIOInfoStructure()
+    io_info_initialize.initialize(basic_io_structure.initial())
+    user_info = resource_op.create_users(no_of_users_to_create=config.user_count)
+    for user in user_info:
+        user_name = user["user_id"]
+
+        ip_and_port = s3cmd_reusable.get_rgw_ip_and_port()
+        s3_auth.do_auth(user, ip_and_port)
+
+        if config.header_size:
```
Comment (psathyan) [inline rgw/v2/tests/s3cmd/test_header_size.py:59]: we could either return if `header_size` is not provided or use a method holding the below lines. It helps with the visuals... also we can improve readability...

**Review by psathyan — COMMENTED** [review]
Looks good... minor queries

File: rgw/v2/tests/s3cmd/test_header_size.py (hunk 3 above)
Comment (anrao19) [reply rgw/v2/tests/s3cmd/test_header_size.py]: Updated as per comment

**Review by psathyan — APPROVED** [review]

**Pull request merged**